apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/template/TemplateReader.java

```diff
@@ -0,0 +1,96 @@
+import org.springframework.stereotype.Service;
+
+import gov.cdc.nbs.questionbank.entity.WaTemplate;
+import gov.cdc.nbs.questionbank.entity.repository.WaTemplateRepository;
+import gov.cdc.nbs.questionbank.template.request.TemplateSearchRequest;
+import gov.cdc.nbs.questionbank.template.response.Template;
+
+@Service
+public class TemplateReader {
+
+	@Autowired
+	private WaTemplateRepository templateRepository;
+
+	public Page<Template> findAllTemplates(Pageable pageable) {
+		Page<WaTemplate> templates = templateRepository.findAll(pageable);
+		List<Template> results = toTempate(templates);
+		return new PageImpl<>(results, pageable, templates.getTotalElements());
+	}
+
+	public Page<Template> searchTemplate(TemplateSearchRequest search, Pageable pageable) {
+
+		List<Template> pageResult = new ArrayList<>();
+		if (simpleSearch(search)) {
+			Optional<WaTemplate> result = templateRepository.findByIdAndTemplateTypeIn(search.getId(),
+					search.getTemplateType());
+			if (result.isPresent()) {
+				pageResult.add(convertTemplate(result.get()));
+			}
+			return new PageImpl<>(pageResult, pageable, pageResult.size());
+
+		} else {
+			Page<WaTemplate> result= templateRepository.searchTemplate(search.getId(), search.getTemplateNm(),
+					search.getConditionCd(), search.getDataMartNm(), search.getRecordStatusCd(),
+					search.getTemplateType(), pageable);
+			pageResult = toTempate(result);
+			
+			return new PageImpl<>(pageResult, pageable, result.getTotalElements());
+
+		}
+	}
+
+	public boolean simpleSearch(TemplateSearchRequest search) {
+		return ((search.getId() > 0l && search.getTemplateType() != null) && search.getConditionCd() == null
+				&& search.getDataMartNm() == null && search.getRecordStatusCd() == null
+				&& search.getTemplateNm() == null) ? true : false;
+	}
+
+	private List<Template> toTempate(Page<WaTemplate> templates) {
+		List<Template> results = new ArrayList<>();
+
+		for (WaTemplate aWaTemplate : templates.getContent()) {
+			results.add(convertTemplate(aWaTemplate));
+		}
+
+		return results;
+	}
+
+	private Template convertTemplate(WaTemplate aWaTemplate) {
+		return new Template(
+				aWaTemplate.getId(),
+				aWaTemplate.getTemplateNm(),
+				aWaTemplate.getTemplateType(),
+				aWaTemplate.getXmlPayload(),
+				aWaTemplate.getPublishVersionNbr(),
+				aWaTemplate.getFormCd(),
+				aWaTemplate.getConditionCd(),
+				aWaTemplate.getBusObjType(),
+				aWaTemplate.getDatamartNm(),
+				aWaTemplate.getRecordStatusCd(),
+				aWaTemplate.getRecordStatusTime(),
+				aWaTemplate.getLastChgTime(),
+				aWaTemplate.getLastChgUserId(),
+				aWaTemplate.getLocalId(),
+				aWaTemplate.getDescTxt(),
+				aWaTemplate.getPublishIndCd(),
+				aWaTemplate.getAddTime(),
+				aWaTemplate.getAddUserId(),
+				aWaTemplate.getNndEntityIdentifier(),
+				aWaTemplate.getParentTemplateUid(),
+				aWaTemplate.getSourceNm(),
+				aWaTemplate.getTemplateVersionNbr(),
+				aWaTemplate.getVersionNote()
```

Comment:
May not need to return all of these fields. Would limit to just whats necessary for now, especially the XmlPayload -- this can be huge

Reply:
@mpeels  updated to a subset of fields needed for the list template dropdown and the manage template library view table